### PR TITLE
feat!: remove v1 Enterprise API support

### PR DESCRIPTION
**Pull request by bradleyjones**
BREAKING CHANGE: no longer compatible with Anchore Enterprise < 4.9

Signed-off-by: Bradley Jones <bradley.jones@anchore.com>

### CODE REVIEW

File: src/main/java/com/anchore/jenkins/plugins/anchore/BuildWorker.java [hunk 1]
```diff
@@ -237,19 +237,7 @@ private void runAnalyzerEngine() throws AbortException {
...
-            }
-            if (null != config.getAnnotations() && !config.getAnnotations().isEmpty()) {
-              JSONObject annotations = new JSONObject();
-              for (Annotation a : config.getAnnotations()) {
-                annotations.put(a.getKey(), a.getValue());
-              }
-              jsonBody.put("annotations", annotations);
-            }
-
+			throw new AbortException("Requires Anchore Enterprise v2 API that can be found in Anchore Enterprise >= 4.9");
```
Comment (juanjsebgarcia) [inline src/main/java/com/anchore/jenkins/plugins/anchore/BuildWorker.java:240]: Tabs??

File: src/main/java/com/anchore/jenkins/plugins/anchore/BuildWorker.java [hunk 2]
```diff
@@ -296,14 +284,8 @@ private void runAnalyzerEngine() throws AbortException {
...
-              // In v1 API this is a list
-              if (config.getEngineApiVersion() == API_VERSION.v1) {
-                JSONArray respJson = JSONArray.fromObject(responseBody);
-                imageDigest = JSONObject.fromObject(respJson.get(0)).getString(Util.GET_VERSION_KEY(config.getEngineApiVersion(), "imageDigest"));
-              } else {
-                JSONObject respJson = JSONObject.fromObject(responseBody);
-                imageDigest = JSONObject.fromObject(respJson).getString(Util.GET_VERSION_KEY(config.getEngineApiVersion(), "imageDigest"));
-              }
+              JSONObject respJson = JSONObject.fromObject(responseBody);
+              imageDigest = JSONObject.fromObject(respJson).getString(Util.GET_VERSION_KEY(config.getEngineApiVersion(), "imageDigest"));
```
[the remark this answers is not in the document]
Comment (juanjsebgarcia) [reply src/main/java/com/anchore/jenkins/plugins/anchore/BuildWorker.java:288]: And then delete that function

File: src/main/java/com/anchore/jenkins/plugins/anchore/BuildWorker.java (hunk 1 above)
Comment (bradleyjones) [reply src/main/java/com/anchore/jenkins/plugins/anchore/BuildWorker.java:240]: urghhh... lol

File: src/main/java/com/anchore/jenkins/plugins/anchore/BuildWorker.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (bradleyjones) [reply src/main/java/com/anchore/jenkins/plugins/anchore/BuildWorker.java:288]: Ah yes forgot about that one will do now

**Review by juanjsebgarcia — APPROVED** [review]

**Pull request merged**